cloud/ec2_fleet.go

```diff
@@ -240,15 +240,15 @@ func (m *ec2FleetManager) GetInstanceStatus(ctx context.Context, h *host.Host) (
 
 	instance, err := m.client.GetInstanceInfo(ctx, h.Id)
 	if err != nil {
+		if ec2err, ok := err.(awserr.Error); ok && ec2err.Code() == EC2ErrorNotFound {
```

Comment:
I think we have to use `ec2err, ok := errors.Cause(err).(awserr.Error)` here because if `GetInstanceInfo` errors, it wraps the error in a specially-defined `error` type that has to be unwrapped to get the original EC2 SDK error.

Reply:
The AWS error is wrapped twice, once by retry and then by `GetInstanceInfo.`, so I'm unwrapping it twice now.

Reply:
It shouldn't be necessary to make multiple calls to `errors.Cause`. `errors.Cause` already unwraps the base error recursively.

Reply:
Oh got it. Still getting familiar with Go. Should have read the docs.